Lib/test/test_threading.py

```diff
@@ -1276,6 +1276,29 @@ def do_flush(*args, **kwargs):
         ''')
         assert_python_ok("-c", script)
 
+    @skip_unless_reliable_fork
+    def test_native_id_after_fork(self):
+        script = """if True:
+            import threading
+            import os
+
+            parent_thread_native_id = threading.current_thread().native_id
+            print(parent_thread_native_id, flush=True)
+            assert parent_thread_native_id == threading.get_native_id()
+            if os.fork() == 0:
+                print(threading.current_thread().native_id, flush=True)
+                assert threading.current_thread().native_id == threading.get_native_id()
+            else:
+                assert parent_thread_native_id == threading.current_thread().native_id
```

Comment:
Since you want to assert on the child's output, explicitly wait for the child process.

```python
try:
    ... asserts ...
finally:
    os.waitpid(child_pid)
```